service/src/main/kotlin/fi/espoo/evaka/attachment/AttachmentsController.kt

```diff
@@ -78,14 +79,26 @@ class AttachmentsController(
         return handleFileUpload(db, user, AttachToIncomeStatement(incomeStatementId), file)
     }
 
+    @PostMapping("/messages/{draftId}", consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
+    fun uploadMessageAttachment(
+        db: Database,
+        user: AuthenticatedUser,
+        @PathVariable draftId: MessageDraftId,
+        @RequestPart("file") file: MultipartFile
+    ): AttachmentId {
+        Audit.AttachmentsUploadForMessage.log()
+        accessControl.requirePermissionFor(user, Action.Global.UPLOAD_MESSAGE_ATTACHMENT)
```

Comment:
This should probably be under MessageDraft scope, which would then check that the user owns that draft.
`accessControl.requirePermissionFor(user, Action.MessageDraft.UPLOAD_ATTACHMENT, draftId)`

Reply:
Yup good catch! Initially the attachments were not tied to a draft.